### PR TITLE
mgr/dashboard: User Management E2E tests

### PR DESCRIPTION
**Pull request by adk3798**
Test breadcrumbs for user management pages
Test creating, editing and deleting a user
Test creating, editing and deleting a role

Fixes: https://tracker.ceph.com/issues/41231
Fixes: https://tracker.ceph.com/issues/41232
Fixes: https://tracker.ceph.com/issues/41233

Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

### CODE REVIEW

**Comment by adk3798** [comment]
jenkins test dashboard

**Comment by adk3798** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by votdev** [comment]
Can you please rebase this PR, i stopped testing this PR after manual merging various conflicts in src/rgw more than 8 times.

File: src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    element(by.id('username')).sendKeys(username);
+    element(by.id('password')).sendKeys(password);
+    element(by.id('confirmpassword')).sendKeys(password);
+    element(by.id('name')).sendKeys(name);
+    element(by.id('email')).sendKeys(email);
+
+    // Click the create button and wait for user to be made
+    const createButton = element(by.cssContainingText('button', 'Create User'));
+    this.moveClick(createButton).then(() => {
+      browser.wait(Helper.EC.presenceOf(this.getTableCell(username)), 10000);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts]: Same for other occurrences. 
```suggestion
      browser.wait(Helper.EC.presenceOf(this.getTableCell(username)), Helper.TIMEOUT);
```

File: src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import { $, browser, by, element } from 'protractor';
+import { Helper } from './helper.po';
+import { PageHelper } from './page-helper.po';
+
+export class UserMgmtPageHelper extends PageHelper {
+  pages = {
+    index: '/#/user-management',
+    users: '/#/user-management/users',
+    userscreate: '/#/user-management/users/create',
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts]: Easier to read:
```suggestion
    userCreate: '/#/user-management/users/create',
```

File: src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    roles: '/#/user-management/roles',
+    rolescreate: '/#/user-management/roles/create'
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts]:
```suggestion
    roleCreate: '/#/user-management/roles/create'
```

**Comment by adk3798** [comment]
Updates:
 - Rebased
 - Changed wait times to Helper.TIMEOUT
 - changed page url variables 'userscreate' and 'rolescreate' to 'userCreate' and 'roleCreate' respectively. Should be easier to read

**Comment by adk3798** [comment]
@votdev I tried to rebase. Is this working for you now?

**Comment by adk3798** [comment]
Made some mistakes when updating. Going to fix

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
Please run "npm run fix"

**Comment by adk3798** [comment]
Fixed linting error that was causing make check to fail

**Pull request merged**